src/Cassandra/TypeCodec.cs

```diff
@@ -1000,8 +1000,9 @@ public static byte[] EncodeCounter(int protocolVersion, IColumnInfo typeInfo, ob
 
         public static object DecodeDouble(int protocolVersion, IColumnInfo typeInfo, byte[] value, Type cSharpType)
         {
-            var buffer = (byte[]) value.Clone();
-            Array.Reverse(buffer);
+            var buffer = new byte[8];
+            for (var i = 0; i < 8; ++i)
+                buffer[i] = value[7 - i];
             return BitConverter.ToDouble(buffer, 0);
```

Comment:
It can be fixed:
```csharp
return BitConverter.ToDouble(
     new byte[] { value[7], value[6], value[5], value[4], value[3], value[2], value[1], value[0]}, 0);
```